Taarafo.Core/Services/Processings/PostImpressions/PostImpressionProcessingService.Exceptions.cs

```diff
@@ -14,8 +15,48 @@ namespace Taarafo.Core.Services.Processings.PostImpressions
 {
     public partial class PostImpressionProcessingService
     {
+        private delegate ValueTask<PostImpression> ReturningPostImpressionFunction();
         private delegate IQueryable<PostImpression> ReturningPostImpressionsFunction();
 
+        private async ValueTask<PostImpression> TryCatch(ReturningPostImpressionFunction returningPostImpressionFunction)
+        {
+            try
+            {
+                return await returningPostImpressionFunction();
+            }
+            catch (NullPostImpressionProcessingException nullPostImpressionProcessingException)
+            {
+                throw CreateAndLogValidationException(nullPostImpressionProcessingException);
+            }
+            catch (InvalidPostImpressionProcessingException invalidPostImpressionProcessingException)
+            {
+                throw CreateAndLogValidationException(invalidPostImpressionProcessingException);
+            }
+            catch (PostImpressionValidationException postImpressionValidationException)
+            {
+                throw CreateAndLogDependencyValidationException(postImpressionValidationException);
+            }
```

Comment:
You don't need this catch block.  All the validation on PostImpression is thrown as an `InvalidPostImpressionProcessingException` then catagorised to `PostImpressionValidationException` with the `throw CreateAndLogValidationException(invalidPostImpressionProcessingException);`

Reply:
Done